Add beta support for AES-256 Password Encoding

Zech-Hein opened the pull request:
- [x] I have considered the risk of behavior change or other zero migration impact (https://github.com/OpenLiberty/open-liberty/wiki/Behavior-Changes).
- [x] If this PR fixes an Issue, the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN" (verify `release bug` label if applicable: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions).
- [x] If this PR resolves an external Known Issue (including APARS), the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN".

################################################################################################

NottyCode committed:
Add support for PBKDF2withHmacSHA256 to wlp.encoding.key

Currently Liberty uses PBKDF2withHmacSHA1 which isn't a good
practice anymore since SHA1 is considered weak. Need to implement
support for PBKDF2withHmacSHA256, but without breaking the
ability to read previously encrypted AES encrypted passwords.

Zech-Hein committed:
add BETA support for aes-256 passwordEncoding

Zech-Hein committed:
add unit tests for beta support of AES-256 password encoding

Zech-Hein commented:
#libby
#build

mshah0722 reviewed (dismissed):
LGTM!

Zech-Hein commented:
#build
#libby

Zech-Hein commented:
First PB had issues possibly due to an old problem. I re-based my branch and started another PB

Zech-Hein committed:
increment package version for updated passwordUtil exception message

Zech-Hein commented:
#build
#libby

mshah0722 reviewed (dismissed):
LGTM

Zech-Hein commented:
last PB hit a known unrelated issue: https://github.com/OpenLiberty/open-liberty/issues/29757

running again
#build

contbld commented:
Your personal pipeline request is at https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=49dfd287-1569-4859-96da-14ccbb0db3b8

Target locations of links might be accessible only to IBM employees.

contbld commented:
<h3>Code analysis and actions</h3>
DO NOT DELETE THIS COMMENT.

- 5 product code files were changed.

contbld commented:
https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=49dfd287-1569-4859-96da-14ccbb0db3b8 has errors/failures.

contbld commented:
The build Zech-Hein-30049-20241107-2028
https://wasrtc.hursley.ibm.com:9443/jazz/resource/itemOid/com.ibm.team.build.BuildResult/_OvH7UJ17Ee-M9e5esDPP_g
completed and has errors or failures.

For help analyzing your personal build, go to https://libh-proxy1.fyre.ibm.com/cognitive/buildAnalysis.html?uuid=_OvH7UJ17Ee-M9e5esDPP_g

Zech-Hein commented:
#build
#libby

contbld commented:
Your personal pipeline request is at https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=3b8e1f0a-1eb5-401a-942b-6445986b9be6

Target locations of links might be accessible only to IBM employees.

contbld commented:
Your personal pipeline request is at https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=ce480d0a-4b94-4e48-ab18-1df2b06517bf

Target locations of links might be accessible only to IBM employees.

contbld commented:
<h3>Code analysis and actions</h3>
DO NOT DELETE THIS COMMENT.

- 5 product code files were changed.

contbld commented:
<h3>Code analysis and actions</h3>
DO NOT DELETE THIS COMMENT.

- 5 product code files were changed.

contbld commented:
https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=3b8e1f0a-1eb5-401a-942b-6445986b9be6 completed successfully!

contbld commented:
https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=ce480d0a-4b94-4e48-ab18-1df2b06517bf has errors/failures.

Zech-Hein commented:
#build

contbld commented:
Your personal pipeline request is at https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=772ca768-affa-4ee6-bdb8-a65e002c47de

Target locations of links might be accessible only to IBM employees.

contbld commented:
<h3>Code analysis and actions</h3>
DO NOT DELETE THIS COMMENT.

- 6 product code files were changed.

contbld commented:
The build Zech-Hein-30049-20241109-2359
https://wasrtc.hursley.ibm.com:9443/jazz/resource/itemOid/com.ibm.team.build.BuildResult/_EC7k0J8wEe-kGLEyqlUFUA
completed and has errors or failures.

For help analyzing your personal build, go to https://libh-proxy1.fyre.ibm.com/cognitive/buildAnalysis.html?uuid=_EC7k0J8wEe-kGLEyqlUFUA

contbld commented:
https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=772ca768-affa-4ee6-bdb8-a65e002c47de has errors/failures.

contbld commented:
Your personal pipeline request is at https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=9e2d135c-f401-46be-b507-2aab34e58851

Target locations of links might be accessible only to IBM employees.

contbld commented:
<h3>Code analysis and actions</h3>
DO NOT DELETE THIS COMMENT.

- 7 product code files were changed.

contbld commented:
<h3>Code analysis and actions</h3>
DO NOT DELETE THIS COMMENT.

- 7 product code files were changed.

contbld commented:
https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=2b364b27-80aa-412f-90ed-4126af747ade completed successfully!

mshah0722 reviewed (dismissed):
LGTM!

contbld commented:
https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=9e2d135c-f401-46be-b507-2aab34e58851 has errors/failures.

Zech-Hein commented:
running one more PB
#build

contbld commented:
Your personal pipeline request is at https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=67cf876b-0b6d-43ab-8c2b-e8192888d339

Target locations of links might be accessible only to IBM employees.

contbld commented:
<h3>Code analysis and actions</h3>
DO NOT DELETE THIS COMMENT.

- 7 product code files were changed.

contbld commented:
https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=67cf876b-0b6d-43ab-8c2b-e8192888d339 has errors/failures.

jhanders34 commented on dev/com.ibm.ws.crypto.passwordutil/test/com/ibm/websphere/crypto/PasswordUtilTest.java:
```diff
@@ -133,7 +130,9 @@ public void testUtil() {
     public void testAESEncoding() throws Exception {
-        assertEquals("The password was not decoded correctly", "alternatepwd", PasswordUtil.decode("{aes}AEmVKa+jOeA7pos+sSfpHNmH1MVfwg8ZoV29iDi6I0ZGcov6hSZsAxMhFr91jTSBYQ=="));
+        //TODO remove beta setting once AES-256 is GA
+        System.setProperty("com.ibm.ws.beta.edition", "true");
```
Should a try/finally unset the beta system property so it isn't set for future tests?

jhanders34 replied:
Or better yet a test to show it failing if beta is not set so that your end up testing the exception path as well.  When you remove the beta guard it will make the unit test fail showing tests that need to be updated.

jhanders34 commented on dev/com.ibm.ws.kernel.boot/publish/platform/kernel-1.0.mf:
```diff
@@ -37,7 +37,7 @@ Subsystem-Content: README.TXT; location:="README.TXT"; type="file",
  com.ibm.websphere.appserver.spi.kernel.filemonitor; version="[1,1.1)"; location:="dev/spi/ibm/"; type="jar",
  com.ibm.websphere.appserver.spi.kernel.embeddable; version="[1.1,1.2)"; location:="dev/spi/ibm/"; type="jar",
- com.ibm.websphere.appserver.spi.kernel.service; version="[1.8,1.9)"; location:="dev/spi/ibm/"; type="jar",
+ com.ibm.websphere.appserver.spi.kernel.service; version="[1.9,1.10)"; location:="dev/spi/ibm/"; type="jar",
```
Line 46 in this file also needs to be updated to a new version for the javadoc zip file.

Zech-Hein replied:
good catch

Zech-Hein committed:
add kernel.boot so productInfo class will be available for beta checks

Zech-Hein commented:
updated javadoc version for kernel spi and added kernel.boot to hopefully fix productInfo NoClassDef issue for securityUtility beta check

#build
#libby

contbld commented:
Your personal pipeline request is at https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=3c674582-41ab-455b-be14-783a776b7a44

Target locations of links might be accessible only to IBM employees.

contbld commented:
Your personal pipeline request is at https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=1c6172ed-ae3c-4d39-8699-f9e05f49bd75

Target locations of links might be accessible only to IBM employees.

Zech-Hein commented:
#build
#git.clone.checkout.branch=fix-security-utility-bvt-noClassDef
#git.clone.repository.sshUrl=git@github.ibm.com:Zech-Hein/WS-CD-Open.git
#git.clone.repository.username=Zech-Hein

contbld commented:
Your personal pipeline request is at https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=3504ab9e-d745-4643-9575-a557f180a014

Target locations of links might be accessible only to IBM employees.

mshah0722 reviewed (approved):
LGTM

contbld commented:
<h3>Code analysis and actions</h3>
DO NOT DELETE THIS COMMENT.

- 8 product code files were changed.

contbld commented:
<h3>Code analysis and actions</h3>
DO NOT DELETE THIS COMMENT.

- 8 product code files were changed.

contbld commented:
https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=3c674582-41ab-455b-be14-783a776b7a44 completed successfully!

contbld commented:
https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=3504ab9e-d745-4643-9575-a557f180a014 has errors/failures.

Zech-Hein commented:
running one more PB in case the last one hangs
#build

contbld commented:
Your personal pipeline request is at https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=28269d0d-81cc-4ae6-9136-4e38eafd5cb5

Target locations of links might be accessible only to IBM employees.

contbld commented:
<h3>Code analysis and actions</h3>
DO NOT DELETE THIS COMMENT.

- 8 product code files were changed.

contbld commented:
https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=1c6172ed-ae3c-4d39-8699-f9e05f49bd75 has errors/failures.

Zech-Hein commented:
the 2nd to last PB finished with the BVT failure fixed. The intermittent zOS failure on the build are unrelated to the changes.

contbld commented:
https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?uuid=28269d0d-81cc-4ae6-9136-4e38eafd5cb5 has errors/failures.
